src/PointwiseFunctions/AnalyticData/GrMhd/BondiHoyleAccretion.cpp

```diff
@@ -0,0 +1,250 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/AnalyticData/GrMhd/BondiHoyleAccretion.hpp"
+
+#include <cmath>
+#include <pup.h>
+
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/DataVector.hpp"  // IWYU pragma: keep
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/EquationOfState.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/PolytropicFluid.hpp"  // IWYU pragma: keep
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+/// \cond
+namespace grmhd {
+namespace AnalyticData {
+
+BondiHoyleAccretion::BondiHoyleAccretion(
+    const double bh_mass, const double bh_dimless_spin,
+    const double rest_mass_density, const double flow_speed,
+    const double magnetic_field_strength, const double polytropic_constant,
+    const double polytropic_exponent) noexcept
+    : bh_mass_(bh_mass),
+      bh_spin_a_(bh_mass * bh_dimless_spin),
+      rest_mass_density_(rest_mass_density),
+      flow_speed_(flow_speed),
+      magnetic_field_strength_(magnetic_field_strength),
+      polytropic_constant_(polytropic_constant),
+      polytropic_exponent_(polytropic_exponent),
+      equation_of_state_{polytropic_constant_, polytropic_exponent_},
+      kerr_schild_coords_{bh_spin_a_} {}
+
+void BondiHoyleAccretion::pup(PUP::er& p) noexcept {
+  p | bh_mass_;
+  p | bh_spin_a_;
+  p | rest_mass_density_;
+  p | flow_speed_;
+  p | magnetic_field_strength_;
+  p | polytropic_constant_;
+  p | polytropic_exponent_;
+  p | equation_of_state_;
+  p | kerr_schild_coords_;
+}
+
+template <typename DataType>
+typename hydro::Tags::SpatialVelocity<DataType, 3>::type
+BondiHoyleAccretion::spatial_velocity(const DataType& r_squared,
+                                      const DataType& sin_theta_squared) const
+    noexcept {
+  auto result =
+      make_with_value<typename hydro::Tags::SpatialVelocity<DataType, 3>::type>(
+          r_squared, 0.0);
+
+  const DataType cos_theta_squared = 1.0 - sin_theta_squared;
+  const DataType sigma = r_squared + square(bh_spin_a_) * cos_theta_squared;
+
+  get<0>(result) = flow_speed_ * sqrt(cos_theta_squared) /
+                   sqrt(1.0 + 2.0 * bh_mass_ * sqrt(r_squared) / sigma);
+  get<1>(result) = -flow_speed_ * sqrt(sin_theta_squared) / sqrt(sigma);
+  // get<2>(result) is identically zero
+
+  return result;
+}
+
+template <typename DataType>
+typename hydro::Tags::MagneticField<DataType, 3>::type
+BondiHoyleAccretion::magnetic_field(const DataType& r_squared,
+                                    const DataType& sin_theta_squared) const
+    noexcept {
+  const double a_squared = bh_spin_a_ * bh_spin_a_;
+  const DataType cos_theta_squared = 1.0 - sin_theta_squared;
+  const DataType cos_theta = sqrt(cos_theta_squared);
```

Comment:
This is off by a sign for z < 0.  In general I'm worried about numerical cancellation with the trig in these functions, so just computing sine and cosine separately might be the right way to go here.